### PR TITLE
Encode and decode time.Duration fields (#201)

### PR DESCRIPTION
**Pull request by quentinmit**
This uses `time.ParseDuration` and `time.Duration.String`, and is
compatible with gopkg.in/yaml's encoding.

### CODE REVIEW

File: decode.go [hunk 1]
```diff
@@ -888,6 +894,34 @@ func (d *Decoder) decodeTime(ctx context.Context, dst reflect.Value, src ast.Nod
...
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to parse duration")
+	}
+	return t, nil
+}
+
+func (d *Decoder) decodeDuration(ctx context.Context, dst reflect.Value, src ast.Node) error {
+	t, err := d.castToDuration(src)
+	if err != nil {
+		return err
```
Comment (goccy) [inline decode.go:919]:
```suggestion
		return errors.Wrapf(err, "failed to convert to duration")
```

File: encode.go [hunk 2]
```diff
@@ -566,6 +572,14 @@ func (e *Encoder) encodeTime(v time.Time, column int) ast.Node {
 	return ast.String(token.New(value, value, e.pos(column)))
 }
 
+func (e *Encoder) encodeDuration(v time.Duration, column int) ast.Node {
+	value := v.String()
+	if e.isJSONStyle {
+		value = strconv.Quote(value)
```
Comment (goccy) [inline encode.go:578]: Could you please add the test case for this branch ?

**Review by goccy — COMMENTED** [review]
Thank you for the great PR ! I commented on some points .

File: decode.go (hunk 1 above)
Comment (quentinmit) [reply decode.go:919]: This is already wrapped (if necessary) inside `castToDuration`

Comment (goccy) [reply decode.go:919]: Yes! That's right. However, we still need to wrap the `err` in order to get the stack trace printed correctly.

Comment (quentinmit) [reply decode.go:919]: Hm, are you saying I should wrap it twice? I'm currently matching the existing implementation of `decodeTime`, which also doesn't wrap the error a second time.

Comment (goccy) [reply decode.go:919]: Yes, it's need to improve debuggability ( please try to output error with `%+v` )
Oh, decodeTime's case is wrong. It's need to wrap...

Comment (quentinmit) [reply decode.go]: I added the call to `Wrapf` on both `decodeDuration` and `decodeTime`. I also added a test case so you can see the resulting messages. It looks like in most cases the stack trace is getting lost anyway at line 1059 when the error is unwrapped before being returned.

**Comment by goccy** [comment]
The latest test has failed. Probably this issue will be fixed by merging the master branch for tracking upstream.

**Comment by quentinmit** [comment]
> The latest test has failed. Probably this issue will be fixed by merging the master branch for tracking upstream.

Would you prefer a merge commit or a rebase?

**Comment by goccy** [comment]
> Would you prefer a merge commit or a rebase?

Either of them is ok 👍

**Comment by quentinmit** [comment]
Sorry for the delay, I merged in the current masster.

**Pull request merged**